plugins/azure/fix_plugin_azure/azure_client.py

```diff
@@ -119,6 +123,10 @@ class RestApiSpec:
     access_path: Optional[str] = None
     expect_array: bool = False
     expected_error_codes: List[str] = field(factory=list)
+    expected_error_codes_with_hints: Dict[str, str] = field(factory=dict)
```

Comment:
Consider defining only one:
```
expected_error_codes: Dict[str, Optional[str]] = field(factory=dict)
```
You can define only the code or the code with error hint.